lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -87,22 +87,29 @@ defmodule Membrane.MP4.Muxer.CMAF do
 
   > ## Note
   > If a stream contains non-key frames (like H264 P or B frames), they should be marked
-  > with a `mp4_payload: %{key_frame?: false}` metadata entry.
+  > with a `h264: %{key_frame?: false}` metadata entry.
   """
   use Membrane.Filter
 
   require Membrane.Logger
 
   alias __MODULE__.{Header, Segment, DurationRange, SegmentHelper}
-  alias Membrane.Buffer
-  alias Membrane.MP4.Payload.AVC1
+  alias Membrane.{Buffer, H264}
   alias Membrane.MP4.{Helper, Track}
   alias Membrane.MP4.Muxer.CMAF.TrackSamplesQueue, as: SamplesQueue
 
   def_input_pad :input,
     availability: :on_request,
     demand_unit: :buffers,
-    accepted_format: Membrane.MP4.Payload
+    accepted_format:
+      any_of(
+        %Membrane.AAC{config: {:esds, _esds}},
+        %Membrane.H264{
+          stream_structure: {:avc1, _dcr},
```

Comment:
I think we need `:avc3` here (CMAF is used in HLS, so the parameters should be stored in-band)

Reply:
I think we should accept both in that case, since avc3 can also have all parameter sets in the dcr

Reply:
I would write it with the use of `when`: `%Membrane.H264{stream_structure: {avc, _dcr}} when avc in `[:avc1, :avc3]`. In the PR with parser I have also suggested adding `is_avc` guard, so perhaps it would be more clear to make that guard public and use in this case?